java/src/main/native/include/jni_utils.hpp

```diff
@@ -747,25 +748,21 @@ inline jthrowable cuda_exception(JNIEnv *const env, cudaError_t status, jthrowab
     return NULL;
   }
 
-  jstring msg = env->NewStringUTF(cudaGetErrorString(status));
-  if (msg == NULL) {
+  const char *err_name = cudaGetErrorName(status);
+  if (err_name == nullptr) {
     return NULL;
   }
+  const char *err_string = cudaGetErrorString(status);
 
-  jobject ret = env->NewObject(ex_class, ctor_id, msg, cause);
-  return (jthrowable)ret;
-}
+  // Build the error message in the format of cudf::cuda_error, so that cudf::jni::CUDA_ERROR_CLASS
+  // can parse both of them.
+  std::string n_msg = "CUDA error encountered at: " + std::string{file} + ":" +
```

Comment:
I wouldn't count on that contents of that exception message being stable.

Reply:
I'd rather see `cudf::cuda_error` updated to allow the extraction of the CUDA error ID rather than relying on parsing.  In general we should be moving away from string-scraping for error identification, not adding more instances of it.

Reply:
Filed https://github.com/rapidsai/cudf/issues/10553